src/ruby_version.rs

```diff
@@ -123,19 +136,29 @@ macro_rules! get_stack_trace(
     ($thread_type:ident) => (
 
         use initialize::StackFrame;
+        use ruby_version::unknown_c_function;
 
         pub fn get_stack_trace<T>(
             ruby_current_thread_address_location: usize,
             source: &T,
             ) -> Result<Vec<StackFrame>, MemoryCopyError> where T: CopyAddress{
+
             let current_thread_addr: usize =
                 copy_struct(ruby_current_thread_address_location, source)?;
             let thread: $thread_type = copy_struct(current_thread_addr, source)?;
+            if stack_field(&thread) as usize == 0 {
+                return Ok(vec!(unknown_c_function()));
+            }
             let mut trace = Vec::new();
             let cfps = get_cfps(thread.cfp as usize, stack_base(&thread) as usize, source)?;
             for cfp in cfps.iter() {
-                if cfp.iseq as usize == 0  || cfp.pc as usize == 0 {
-                    debug!("Either iseq or pc was 0, skipping CFP");
+                if cfp.iseq as usize == 0 {
+                    // this means this stack frame is a C function, note that and continue
```

Comment:
is this 100% true verified by reading mri source? or is it an assumption you're making?

Reply:
the reason I believe this is that I grepped the Ruby source for `vm_push_frame` calls that set the `iseq` argument to 0 and they all have VM_FRAME_FLAG_CFRAME in them. I don't think I can check for that flag directly. 

```
vm.c:    vm_push_frame(th, 0 /* dummy iseq */, VM_FRAME_MAGIC_DUMMY | VM_ENV_FLAG_LOCAL | VM_FRAME_FLAG_FINISH | VM_FRAME_FLAG_CFRAME /* dummy frame */,
vm_eval.c:	vm_push_frame(th, 0, VM_FRAME_MAGIC_CFUNC | VM_FRAME_FLAG_CFRAME | VM_ENV_FLAG_LOCAL, recv,
vm_insnhelper.c:    vm_push_frame(th, 0, VM_FRAME_MAGIC_CFUNC | VM_FRAME_FLAG_CFRAME | VM_ENV_FLAG_LOCAL,
vm_insnhelper.c:    vm_push_frame(th, NULL, VM_FRAME_MAGIC_CFUNC | VM_FRAME_FLAG_CFRAME | VM_ENV_FLAG_LOCAL, recv,

```